### PR TITLE
Remove support for `new borrowed`

### PR DESCRIPTION
**Pull request by jabraham17**
Removes support for creating a class object as `new borrowed`, deprecated in https://github.com/chapel-lang/chapel/pull/20589.

Although Chapel has special parser support for this, this PR does not remove it, instead opting to keep it to have better error messages as a post parse check. This also maintains a check during resolution for calls where a type alias resolves to `borrowed`.

Testing
- [x] local testing of tests affected
- [x] dyno tests
- [x] paratest with comm=none
- [x] paratest with comm=gasnet

[Reviewed by @lydia-duncan]

### CODE REVIEW

File: test/classes/ferguson/class-double-modifier.good [hunk 1]
```diff
@@ -34,6 +34,3 @@ class-double-modifier.chpl:24: error: duplicate decorators - shared shared A(int
 class-double-modifier.chpl:23: In function 'doit3':
 class-double-modifier.chpl:24: error: duplicate decorators - shared borrowed A(int(64))
   class-double-modifier.chpl:58: called as doit3(type C = borrowed A(int(64)))
-class-double-modifier.chpl:30: In function 'doit4':
```
Comment (lydia-duncan) [inline test/classes/ferguson/class-double-modifier.good:37]: I'm surprised more wasn't removed from this .good file, given the lines you removed.

File: test/associative/types/testNilableBorrowed.chpl [hunk 2]
```diff
@@ -1,15 +0,0 @@
...
-
-D += 1;
-
-var A: [D] borrowed T?;
-var a = new T();
-var b = a.borrow();
-
-A[1] = b;
-
-assert(A.size == 1);
```
Comment (lydia-duncan) [inline test/associative/types/testNilableBorrowed.chpl:15]: How did this test behave after your compiler changes?  It kinda seems like something we want to support, so it'd be nice if the test would stick around.

**Review by lydia-duncan — COMMENTED** [review]
Couple of questions (though I suspect one of them will be handled after a test run)

File: test/associative/types/testNilableBorrowed.chpl (hunk 2 above)
Comment (jabraham17) [reply test/associative/types/testNilableBorrowed.chpl:15]: This one still compiles cleanly, I can reinstate it

File: test/classes/ferguson/class-double-modifier.good (hunk 1 above)
Comment (jabraham17) [reply test/classes/ferguson/class-double-modifier.good:37]: Ah you are right, this test does need more changes, I guess I missed this one in my local testing

**Review by lydia-duncan — APPROVED** [review]
Looks good, thanks!

**Pull request merged**